pythainlp/util/thai_lunar_date.py

```diff
@@ -0,0 +1,828 @@
+        Fyear, FDev = StartY[32][0], StartY[32][1]
+    elif iYear >= 2051:
+        Fyear, FDev = StartY[31][0], StartY[31][1]
+    elif iYear >= 2046:
+        Fyear, FDev = StartY[30][0], StartY[30][1]
+    elif iYear >= 2041:
+        Fyear, FDev = StartY[29][0], StartY[29][1]
+    elif iYear >= 2036:
+        Fyear, FDev = StartY[28][0], StartY[28][1]
+    elif iYear >= 2031:
+        Fyear, FDev = StartY[27][0], StartY[27][1]
+    elif iYear >= 2026:
+        Fyear, FDev = StartY[26][0], StartY[26][1]
+    elif iYear >= 2021:
+        Fyear, FDev = StartY[25][0], StartY[25][1]
+    elif iYear >= 2016:
+        Fyear, FDev = StartY[24][0], StartY[24][1]
+    elif iYear >= 2011:
+        Fyear, FDev = StartY[23][0], StartY[23][1]
+    elif iYear >= 2006:
+        Fyear, FDev = StartY[22][0], StartY[22][1]
+    elif iYear >= 2001:
+        Fyear, FDev = StartY[21][0], StartY[21][1]
+    elif iYear >= 1996:
+        Fyear, FDev = StartY[20][0], StartY[20][1]
+    elif iYear >= 1991:
+        Fyear, FDev = StartY[19][0], StartY[19][1]
+    elif iYear >= 1986:
+        Fyear, FDev = StartY[18][0], StartY[18][1]
+    elif iYear >= 1981:
+        Fyear, FDev = StartY[17][0], StartY[17][1]
+    elif iYear >= 1976:
+        Fyear, FDev = StartY[16][0], StartY[16][1]
+    elif iYear >= 1971:
+        Fyear, FDev = StartY[15][0], StartY[15][1]
+    elif iYear >= 1966:
+        Fyear, FDev = StartY[14][0], StartY[14][1]
+    elif iYear >= 1961:
+        Fyear, FDev = StartY[13][0], StartY[13][1]
+    elif iYear >= 1956:
+        Fyear, FDev = StartY[12][0], StartY[12][1]
+    elif iYear >= 1951:
+        Fyear, FDev = StartY[11][0], StartY[11][1]
+    elif iYear >= 1946:
+        Fyear, FDev = StartY[10][0], StartY[10][1]
+    elif iYear >= 1941:
+        Fyear, FDev = StartY[9][0], StartY[9][1]
+    elif iYear >= 1936:
+        Fyear, FDev = StartY[8][0], StartY[8][1]
+    elif iYear >= 1931:
+        Fyear, FDev = StartY[7][0], StartY[7][1]
+    elif iYear >= 1926:
+        Fyear, FDev = StartY[6][0], StartY[6][1]
+    elif iYear >= 1921:
+        Fyear, FDev = StartY[5][0], StartY[5][1]
+    elif iYear >= 1916:
+        Fyear, FDev = StartY[4][0], StartY[4][1]
+    elif iYear >= 1911:
+        Fyear, FDev = StartY[3][0], StartY[3][1]
+    elif iYear >= 1906:
+        Fyear, FDev = StartY[2][0], StartY[2][1]
+    elif iYear >= 1901:
+        Fyear, FDev = StartY[1][0], StartY[1][1]
+    else:
+        Fyear, FDev = StartY[0][0], StartY[0][1]
+
+    return Fyear, FDev
+
+
+def AthikaMas(iYear: int) -> bool:
+    Athi = XLMod((iYear - 78) - 0.45222, 2.7118886)
+    return Athi < 1
+
+
+def AthikaVar(iYear: int) -> bool:
+    if AthikaMas(iYear):
+        return False
+    if AthikaMas(iYear + 1):
+        CutOff = 1.69501433191599E-02
+    else:
+        CutOff = -1.42223099315486E-02
+    return Deviation(iYear) > CutOff
+
+
+def Deviation(iYear: int) -> float:
+    FDev = None
+    Fyear = None
+    CurrDev = 0
+    lastDev = None
```

Comment:
```suggestion
    curr_dev = 0.0
    last_dev = 0.0
```

Since Python type checkers are more common now, avoid initialize floats with `None`.
`None` is compatible with neither int nor float.

In this context, lines 399-400 can be removed as values can be initialize by the `calculate_Fyear_FDev()` function. And since both `CurrDev` and `lastDev` are float, we can put 0.0 (cannot be 0, as that will be an int).

Reply:
Done